src/main/java/org/isf/dicom/gui/DicomGui.java

```diff
@@ -266,126 +264,121 @@ private void addEventListener() {
-						new SourceFiles(dummyFileDicom, selectedFile, patient, numfiles, thumbnail, new DicomLoader(numfiles, myJFrame));
-					} else {
-						// single file 
-						try {
-							SourceFiles.loadDicom(dummyFileDicom, selectedFile, patient);
-						} catch (Exception e1) {
-							OHServiceExceptionUtil.showMessages((OHDicomException) e1);
-						}
-						thumbnail.initialize();
+				if (selectedFile.isDirectory()) {
+					//folder
+					thumbnail.disableLoadButton();
+					new SourceFiles(dummyFileDicom, selectedFile, patient, numfiles, thumbnail, new DicomLoader(numfiles, myJFrame));
+				} else {
+					// single file
+					try {
+						SourceFiles.loadDicom(dummyFileDicom, selectedFile, patient);
+					} catch (Exception e1) {
+						OHServiceExceptionUtil.showMessages((OHDicomException) e1);
 					}
+					thumbnail.initialize();
 				}
 			}
 		});
 	}
 
 	private void actionListenerJButtonDeleteDicom() {
-		jButtonDeleteDicom.addActionListener(new ActionListener() {
-
-			public void actionPerformed(ActionEvent e) {
+		jButtonDeleteDicom.addActionListener(e -> {
 
-				Object[] options = { MessageBundle.getMessage("angal.dicom.delete.yes"), MessageBundle.getMessage("angal.dicom.delete.no") };
+			Object[] options = { MessageBundle.getMessage("angal.dicom.delete.yes"), MessageBundle.getMessage("angal.dicom.delete.no") };
 
-				int n = JOptionPane.showOptionDialog(DicomGui.this, MessageBundle.getMessage("angal.dicom.delete.request"),
-						MessageBundle.getMessage("angal.dicom.delete.title"),
-						JOptionPane.YES_NO_CANCEL_OPTION, JOptionPane.QUESTION_MESSAGE, null, options, "");
+			int n = JOptionPane.showOptionDialog(DicomGui.this, MessageBundle.getMessage("angal.dicom.delete.request"),
+					MessageBundle.getMessage("angal.dicom.delete.title"),
+					JOptionPane.YES_NO_CANCEL_OPTION, JOptionPane.QUESTION_MESSAGE, null, options, "");
 
-				if (n == 0) {
-					try {
-						DicomManagerFactory.getManager().deleteSerie(patient, thumbnail.getSelectedInstance().getDicomSeriesNumber());
-					} catch (OHServiceException ohServiceException) {
-						MessageDialog.showExceptions(ohServiceException);
-					}
+			if (n == 0) {
+				try {
+					DicomManagerFactory.getManager().deleteSerie(patient, thumbnail.getSelectedInstance().getDicomSeriesNumber());
+				} catch (OHServiceException ohServiceException) {
+					MessageDialog.showExceptions(ohServiceException);
 				}
-				thumbnail.initialize();
-				//selectedElement = null;
-				//detail();
-				((DicomViewGui) jPanelDetail).clear();
 			}
+			thumbnail.initialize();
+			//selectedElement = null;
+			//detail();
+			((DicomViewGui) jPanelDetail).clear();
 		});
 	}
 
 	private void actionListenerjButtonCloseDicom() {
-		jButtonCloseDicom.addActionListener(new ActionListener() {
-
-			public void actionPerformed(ActionEvent e) {
-				System.exit(100);
-				// this.dispose();
-			}
+		jButtonCloseDicom.addActionListener(e -> {
+			System.exit(100);
+			// this.dispose();
```

Comment:
`jButtonCloseDicom` is never added to the gui, I think we can remove it

Reply:
It has been removed

Reply:
Thanks!